Sources/Purchasing/StoreKit2/StoreKit2PurchaseIntentListener.swift

```diff
@@ -157,7 +157,13 @@ struct StorePurchaseIntent: Sendable, Equatable {
 @available(tvOS, unavailable)
 @available(watchOS, unavailable)
 @available(visionOS, unavailable)
-protocol StoreKit2PurchaseIntentType: Equatable, Identifiable, Sendable {
+protocol StoreKit2PurchaseIntentType: Equatable, Sendable {
+
+    // WARNING: **DO NOT** make this type conform to Identifiable!!!
+    // While StoreKit.PurchaseIntent conforms to Identifiable in iOS 18+,
+    // the conformance is not available in earlier versions of iOS, and this will
+    // cause a runtime crash when trying to typecast a StoreKit.PurchaseIntent
+    // to a StoreKit2PurchaseIntentType in iOS 16.4..<18.0.
```

Comment:
let's add a link to this PR or the GH issue

Reply:
Added in a4f6956d294056960d10d9fc3cf9f1596f938b7e